src/generic/toast-context/hooks.js

```diff
@@ -0,0 +1,25 @@
+// @ts-check
+import React from 'react';
+
+const useToastContext = () => {
+  // TODO, We can convert this to a queue of messages,
+  // see: https://github.com/open-craft/frontend-app-course-authoring/pull/38#discussion_r1638990647
+
+  const [toastMessage, setToastMessage] = React.useState(/** @type{null|string} */ (null));
+
+  React.useEffect(() => () => {
+    // Cleanup function to avoid updating state on unmounted component
+    setToastMessage(null);
+  }, []);
+
+  const showToast = React.useCallback((message) => setToastMessage(message), [setToastMessage]);
+  const closeToast = React.useCallback(() => setToastMessage(null), [setToastMessage]);
```

Comment:
Nit: you don't have to include `setToastMessage` in the dependencies array, because React guarantees that the setter function of a `useState` hook will never change. But I guess it's fine to leave it if you want.

Reply:
Updated here: https://github.com/open-craft/frontend-app-course-authoring/pull/38/commits/853bdf62219f077ea20f4de1213d14b847c3ac52#diff-803ee2797c9b110536f16506480a9d55418bae1cf42dfec7806b9027298691f7R18